go.mod

```diff
@@ -4,8 +4,8 @@ go 1.18
 
 require (
 	github.com/Azure/azure-sdk-for-go v63.1.0+incompatible
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.7.1
-	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.1
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.6.1
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.0
```

Comment:
Is it needed? @kimorris27 is currently working towards moving the ARO operator to a chained credential which may require this version of `azidentity`.

Reply:
I think we can avoid changing the version in this PR. I cloned this branch and changed `azidentity` back to `v1.3.1`, and a `go mod tidy` didn't change anything else aside from setting `azcore` back to `v1.7.1`.

@schiruma, would you mind changing the versions for `azcore` and `azidentity` back to the values on the master branch and then making sure your code continues to work? I don't think using 1.3.0 instead of 1.3.1 will cause problems, but it would be nice to stick with the newer one since I had already updated us to it. Might as well keep those patch contents added in 1.3.1.

Reply:
@kimorris27 made changes as mentioned.